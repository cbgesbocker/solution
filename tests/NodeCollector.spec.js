const NodeCollector = require("../src/lib/NodeCollector");

describe("NodeCollector.js", () => {
  let nodeCollector, getNodeSpy;

  beforeEach(async () => {
    getNodeSpy = jest.fn(async (id) => ({
      data: [
        {
          id,
          child_node_ids: [],
        },
      ],
    }));
    nodeCollector = new NodeCollector({ getNode: getNodeSpy }, "12345");
  });

  test("NodeCollector should call getNode on loading", async () => {
    await nodeCollector.load();
    expect(getNodeSpy).toHaveBeenCalled();
  });

  test("NodeCollector should call getNode with the test ID", async () => {
    await nodeCollector.load();
    expect(getNodeSpy).toHaveBeenCalledWith("12345");
  });
});
