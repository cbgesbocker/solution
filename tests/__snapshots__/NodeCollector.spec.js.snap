// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeCollector.js NodeCollector should call getNode with the test ID 1`] = `
NodeCollector {
  "client": Object {
    "getNode": [MockFunction] {
      "calls": Array [
        Array [
          "12345",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Promise {},
        },
      ],
    },
  },
  "rnode": Object {
    "child_node_ids": Array [],
    "id": "12345",
  },
  "rootNodeId": "12345",
  "tree": NodeTree {
    "getNode": [MockFunction] {
      "calls": Array [
        Array [
          "12345",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Promise {},
        },
      ],
    },
    "max": Object {
      "count": 0,
      "node": Object {},
    },
    "nodeCountDict": Object {
      "12345": 1,
    },
    "nodeDict": Object {
      "12345": Object {
        "child_node_ids": Array [],
        "id": "12345",
      },
    },
    "rootNode": Object {
      "child_node_ids": Array [],
      "id": "12345",
    },
    "visited": Array [
      "12345",
    ],
  },
}
`;
